core/ir/instr_api.h

```diff
@@ -1439,12 +1439,19 @@ DR_API
  * the top half while others zero it when writing to the bottom half).
  * This zeroing will occur even if \p instr is predicated (see instr_is_predicated()).
  */
-/* XXX i#1312: For AVX-512, we will want a instr_zeroes_zmmh function as well that also
- * includes the vzeroupper instruction.
```

Comment:
Could you confirm that vzeroupper returns true?

Reply:
Nice catch! Turns out that the ir for vzeroupper and vzeroall do not include the simd registers as operands. This needs to be fixed, if possible. We need to include them even if there are implicit. Think push instruction which has an implicit operand related to the stack that our ir recognises.

Reply:
Done.